src/metabase/util/malli/fn.clj

```diff
@@ -295,8 +295,38 @@
   `(let [~'&f ~(deparameterized-fn-form parsed)]
      (core/fn ~@(instrumented-fn-tail error-context (fn-schema parsed)))))
 
+(defn uninstrumented-fn-form
+  "Given a `fn-tail` like
+
+  ([x :- :int y] (+ 1 2))
+
+  and parsed by [[parsed-fn-tail]],
+
+  return an unevaluated, and uninstrumented fn form like
+
+  (fn [x y] (+ 1 2))
+
+  This is used to emit fn which ignore checks."
+  [parsed]
+  (deparameterized-fn-form parsed))
+
+;; ------------------------------ Skipping Namespace Enforcement in prod ------------------------------
+
+;; FIXME: is there a cleaner way to setup `namespaces-toskip`?
+;; I was thinking we could `swap!` it from the namespaces themselves with:
+;; `(swap namespaces-toskip conj *ns*)`
+(def namespaces-toskip
```

Comment:
something like this might be better:

```clojure
(defn instrument-ns?
  "Used to track namespaces to not enforce malli schemas on with `mu.fn/fn`."
  [namespace]
  (let [lib-and-middleware [#"^metabase\.lib\..*"
                            #"^metabase\.query-processor\.middleware\..*"]
        matches?    (fn [namespace regexes]
                      (let [n (-> namespace ns-name str)]
                        (some #(re-matches % n) regexes)))
        ad-hoc      '#{metabase.pulse.render} ;; just to demonstrate
        m           (meta namespace)]
    (or (:instrument/always m)
        (true? (:instrument/never m))
        (not (matches? namespace lib-and-middleware))
        (not (contains? ad-hoc (ns-name namespace))))))
```